model-optimizer/extensions/middle/ConvertGroupedStridedSlice.py

```diff
@@ -30,6 +31,73 @@
 from mo.ops.unsqueeze import Unsqueeze
 
 
+def group_by_with_binary_predicate(xs: list, predicate: Callable) -> list:
```

Comment:
Let's move all these grouping functions to a common file like `mo/utils/utils.py`

Reply:
No, `CanBeFused` should not always return `True`. We should use     
```python
         fuser = CanBeFused()
         sequences = group_by_with_binary_predicate(interps, lambda prev, x: is_next(prev, x) and fuser(prev, x))
```
instead of 
```python
        sequences = collect_sequences(interps)
```

Reply:
My point is that we have 2 almost identical functions. They should be refactored and only one function should be kept. The technical details of the implementation is out of scope of my comment.

Reply:
Fixed.